### PR TITLE
[render preview] Add Inkeep to project setup

### PR DESCRIPTION
**Pull request by ysadamt**
## Summary

- Add Inkeep chat button on project setup page
- Add styling to adhere to highlight design system

Fixes GRO-133

## How did you test this change?

Click chat button on setup page, search and ask AI

[Screencap](https://www.loom.com/share/0815dadbcc10458a96f8a6c44ea091a6?sid=e06ca83d-e43c-47b4-9f40-f5460ee41f18)
## Are there any deployment considerations?

No

## Does this work require review from our design team?

Yes

### CODE REVIEW

**Comment by Vadman97** [comment]
@lewisl9029 any clue why the preview might be failing here? https://github.com/highlight/highlight/pull/9357/checks?check_run_id=30553269106

**Comment by lewisl9029** [comment]
Ah yep, the new dependency on next is probably the problem. It's a pretty big and hairy package and a lot of parts assume server-side js so we have some trouble with it by default. But in this case it looks like we're only using the next/dynamic entrypoint here, so I think we can get it working with a bit of config. 

Mind adding this in .reflame.config.jsonc under `npmPackages`?

```
    "next": {
      "entryPoints": [
        "/dynamic"
      ],
    },
```

This will make sure we only attempt to build the `/dynamic` entrypoint for our client bundle, and should work assuming `/dynamic` doesn't assume a server environment.

Alternatively, it sounds like outside of the SSR functionality, next/dynamic is just a thin wrapper around React.lazy and Suspense, so we could always just fallback to using those directly and remove the next dep altogether since we're not using SSR:

```
const ChatButton = React.lazy(
  () => import('@inkeep/uikit').then((mod) => ({ default: mod.InkeepChatButton })),
)

// ...

<Suspense fallback={<div>loading</div>}>
  <ChatButton {...chatButtonProps} />
</Suspense>
```

Let me know if y'all run into any trouble with either of those approaches. Happy to help out further!

File: frontend/src/hooks/useInkeepSettings.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+type InkeepSharedSettings = {
+	baseSettings: InkeepBaseSettings
+	aiChatSettings: InkeepAIChatSettings
+	searchSettings: InkeepSearchSettings
+	modalSettings: InkeepModalSettings
+}
+
+const useInkeepSettings = (): InkeepSharedSettings => {
+	const baseSettings: InkeepBaseSettings = {
+		apiKey: '3721b3f00a2e161d2eb143932695e984a9db05ffb89bdb57',
```
[the remark this answers is not in the document]
Comment (Vadman97) [reply frontend/src/hooks/useInkeepSettings.ts:17]: public frontend token

File: frontend/src/pages/Setup/SetupInkeepChatButton.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { InkeepChatButtonProps } from '@inkeep/uikit'
+import useInkeepSettings from '@/hooks/useInkeepSettings'
+import { lazy, Suspense } from 'react'
+
+const ChatButton = lazy(() =>
+	import('@inkeep/uikit').then((mod) => ({ default: mod.InkeepChatButton })),
```
Comment (Vadman97) [inline frontend/src/pages/Setup/SetupInkeepChatButton.tsx:6]: looks like the chat button isn't visible by default?

https://www.loom.com/share/8a7f2742713d4970bd7c86a202fc6541

**Review by Vadman97 — APPROVED** [review]
looks good other than the visibility issue - will defer to @julian-highlight for additional thoughts on positioning

**Comment by Vadman97** [comment]
@lewisl9029 appreciate your help! didn't notice we started to pull in `next/dynamic`

File: frontend/src/pages/Setup/SetupInkeepChatButton.tsx (hunk 2 above)
Comment (ysadamt) [reply frontend/src/pages/Setup/SetupInkeepChatButton.tsx:6]: hmm... seems to be not showing in the reflame preview deployment, but showing fine locally. the custom styles I added are also not being applied to the modal.

Comment (lewisl9029) [reply frontend/src/pages/Setup/SetupInkeepChatButton.tsx:6]: Just took a look. The custom styles not applying is due to Reflame's lack of support for the public dir (we need everything to be part of your module graph in order to properly cache and invalidate resources).

I believe that can be fixed by moving your css file into /src and then importing it as a url like this:

```
import inkeepCssUrl from '@/inkeep.css'

// ...

stylesheetUrls: [inkeepCssUrl]
```

I'm not quite sure about the `visibility: hidden` thing yet at the moment. Curious if that's also there in your local dev? Maybe something else overrides it that doesn't end up applying in the Reflame preview.

Comment (Vadman97) [reply frontend/src/pages/Setup/SetupInkeepChatButton.tsx:6]: @lewisl9029 thanks again - looks like one trick necessary was the `?url` suffix on the import to make sure the value was the URL and not the css file contents - https://vitejs.dev/guide/assets#explicit-url-imports

**Comment by Vadman97** [comment]
now working everywhere

locally with vite dev 
Screenshot from 2024-09-24 23-44-37
reflame preview
Screenshot from 2024-09-24 23-48-56

**Comment by Vadman97** [comment]
@julian-highlight you can review bot button here: https://preview.highlight.io/1/setup/client/js?%7Er_preview=%7B%22action%22%3A%22start%22%2C%22mode%22%3A%22production%22%2C%22region%22%3A%22ewr%22%2C%22variantId%22%3A%2201J8JEQ22YKHV17SHWY319019G%22%2C%22variantData%22%3A%22%7B%5C%22type%5C%22%3A%5C%22branch%5C%22%2C%5C%22branch%5C%22%3A%5C%22inkeep-search-setup%5C%22%2C%5C%22githubOwnerName%5C%22%3A%5C%22highlight%5C%22%2C%5C%22githubRepositoryName%5C%22%3A%5C%22highlight%5C%22%7D%22%7D

**Comment by Vadman97** [comment]
```
2024-09-25T14:26:39.8305728Z [ERROR] command finished with error: command (/home/runner/actions-runner/_work/highlight/highlight/frontend) /tmp/xfs-55bab713/yarn run build exited (129)
2024-09-25T14:26:39.8319547Z ##[error]@highlight-run/frontend#build: command (/home/runner/actions-runner/_work/highlight/highlight/frontend) /tmp/xfs-55bab713/yarn run build exited (129)
2024-09-25T14:26:39.8319956Z 
2024-09-25T14:26:39.8320079Z  Tasks:    27 successful, 28 total
2024-09-25T14:26:39.8320197Z Cached:    0 cached, 28 total
2024-09-25T14:26:39.8320295Z   Time:    4m16.673s 
2024-09-25T14:26:39.8320491Z Failed:    @highlight-run/frontend#build
2024-09-25T14:26:39.8320501Z 
2024-09-25T14:26:39.8320624Z  ERROR  run failed: command  exited (129)
```

**Pull request merged**